subiquity/controllers/keyboard.py

```diff
@@ -52,14 +53,13 @@ def start_ui(self):
             variant = self.answers.get('variant', '')
             self.done(KeyboardSetting(layout=layout, variant=variant))
 
-    def done(self, setting):
-        self.run_in_bg(
-            lambda: self.model.set_keyboard(setting),
-            self._done)
-
-    def _done(self, fut):
-        log.debug("KeyboardController._done next-screen")
+    async def apply_settings(self, setting):
+        await self.model.set_keyboard(setting)
+        log.debug("KeyboardController next-screen")
         self.signal.emit_signal('next-screen')
 
+    def done(self, setting):
+        asyncio.create_task(self.apply_settings(setting))
```

Comment:
Would you want some subiquity task abstraction here in case we need to do something different than direct calling of asyncio.create_task?

Reply:
Not entirely sure, but this looks like it should use the scheduel_task() method you've defined.

Reply:
Ah yes good catch.